Aptslav data provider

VolodymyrHoliev opened the pull request:
This version of AptslavDataProvider implements DataProvider interface.I haven't merged this branch with DataProvider_enhancement for now.

ritalinch committed:
Master (#1)

* add repositories and controllers

* add converters

* add DS data provider and both DS and Slav DTOs

* add DS data provider and both DS and Slav DTOs

* application.yaml file is up-to-date

* some changes in application.yaml, start of Slav Data Provider

Co-authored-by: Volodymyr Holiev <90215834+VolodymyrHoliev@users.noreply.github.com>

VolodymyrHoliev committed:
Getting started with AptslavDataProvider

VolodymyrHoliev committed:
Refactoring

VolodymyrHoliev committed:
Added DtoMapper

VolodymyrHoliev committed:
Fixes

VolodymyrHoliev committed:
Applied patch

VolodymyrHoliev committed:
Added SchedulerService

VolodymyrHoliev committed:
AptslavDataProvider refactoring

zyoneal committed:
fix script name

VolodymyrHoliev committed:
Added request mappers

VolodymyrHoliev committed:
Clean up

VolodymyrHoliev committed:
Reformat code

VolodymyrHoliev committed:
application-test.yaml fix

VolodymyrHoliev committed:
Added missed Components

pavlosloboda reviewed (approved):
I think you do not need to delete PharmacyController.java here. Except for this, it looks fine to me and can be merged as soon as conflicts are resolved.

VolodymyrHoliev committed:
Reformat code

VolodymyrHoliev commented:
Here in this branch I deleted all my controllers.The point is to make merging this branch with main repository as simple as possible.
I thought I should keep this branch as simple as it possible,because later this branch should be merged into main repository.

VolodymyrHoliev committed:
Fixed V1_1_0__insert_into_pharmacies_aptslav.sql script

VolodymyrHoliev committed:
Added missed blank lines in MedicineDto, DtoToMedicineConverter and DtoToPriceConverter

VolodymyrHoliev committed:
Added missed annotations in AptslavMedicineDto

VolodymyrHoliev committed:
Added property for 'step' variable in AptslavDataProvider

VolodymyrHoliev committed:
Fixed total pages calculation logic in AptslavDataProvider

VolodymyrHoliev committed:
Moved aptslav dto's in corresponding package

VolodymyrHoliev committed:
Added missed web-client configuration

VolodymyrHoliev committed:
Bug fix

VolodymyrHoliev committed:
Deleted redundant SchedulerService.java

VolodymyrHoliev committed:
Deleted redundant converters package

VolodymyrHoliev committed:
Added basic tests for AptslavDataProvider

Anna-Kudiakova committed:
fix rozetka data provider

VolodymyrHoliev committed:
ApiMedicineDtoConverter bug fix

Commit:
Export to Excel file

Commit:
Add advanced search

Anna-Kudiakova committed:
Add integration tests for controllers (#89)

* integration tests for pharmacy controller

* integration tests for medicine controller

* integration tests for price controller

* add blank lines

* replace container tags with non-container

* make constant var upper-case

* make variables non-final

* make values non string literal

MuchosChuchos committed:
Add validation to request body in controllers (#88)

* fix: Add validation to request body in controllers

Closes #86

* refactor: Change @notnull to @notblank

* refactor: Set @NotNull for price field.

* refactor(Add @NoArgs... and @AllArgs... to dtos):

FedenkoMykhailo committed:
Add jacoco plugin configuration

Oleksandr-Eleks committed:
Configure PostgreSQL for CI tests, remove Travis

VolodymyrHoliev committed:
Code refactoring

Oleksandr-Eleks committed:
Rename migration script before merge, formatting

VolodymyrHoliev committed:
AptslavDataProvider refactoring
Added ResponseBodyIsNullException
Added exception handling

VolodymyrHoliev committed:
Changes in AptslavDataProviderIT

VolodymyrHoliev committed:
renamed migration script

VolodymyrHoliev committed:
missing table [advanced_search_view] bug fixed

VolodymyrHoliev committed:
fixed DbUnit tests
Now test aren't deleting all data from tables

VolodymyrHoliev committed:
changes according to SonarCloud checks

VolodymyrHoliev committed:
Added config to make Jacoco ignore generated code

SofiiaOlenyn committed:
Add basic tests for PriceController (#87)

* add tests for PriceController

* refactor PriceControllerTest

kolyaratishin committed:
add data import from csv (#113)

* add data import from csv

* Encapsulate logic of storing to database

* add tests

VolodymyrHoliev committed:
Deleted redundant migration script

VolodymyrHoliev committed:
Reformat code

VolodymyrHoliev committed:
Added api calls limitations

MariamnaKh committed:
remove content type (#119)

Anna-Kudiakova committed:
add limits on requests (#121)

kolyaratishin committed:
export to pdf (#118)

* add data export to pdf

* add test

VolodymyrHoliev committed:
Revert "fixed DbUnit tests"

This reverts commit 6b56caf358481b858719f4642c8c95d1286faf06.

VolodymyrHoliev committed:
Removed redundant blank lines

Added requested blank lines

VolodymyrHoliev committed:
Removed redundant calculateTotalPages() method from AptslavDataProvider

VolodymyrHoliev committed:
Added property for api calls limit

VolodymyrHoliev committed:
Added missed blank line in AptslavDataProviderIT

VolodymyrHoliev committed:
Removed redundant field from AptslavDataProvider

VolodymyrHoliev committed:
Added missed commas in javadoc and missed blank lines

MuchosChuchos committed:
refactor: add limitations of fetching meds (#120)

kolyaratishin committed:
Add data retrieval restrictions (#125)

MuchosChuchos committed:
Handling wrong id exception (#104)

* fix: Handle InvalidIdentifierException

Add new custom exception.
Add handling api exceptions.
Handle invalid id exception in controllers
 in delete, update and getById methods.

Closes #90

* style: Add blank lines

* refactor: change rest controller exception handling

Extend scope of class hierarchy for response entity.
Change annotation.
Shrink methods body.

* style: minor improvements

* fix: process id exception

* fix: change printing exception from System#err to log

* test: Add some unit tests for PharmacyController.

Add methods for testing:
- getAll()
- getById()
- getById() - wrongId
- create()

* test: Cover the PharmacyController

Add methods for testing:
- create() - invalid request body
- delete()
- delete() - wrong ID
- update()
- update() - invalid request body

* refactor: change deleteById test accordingly logic of delete method in Controller

* refactor: change 2 test methods that expect wrong id in PriceControllerTest

-getById_noExistsId_NotFound()
-updatePriceById_noExistsId_NotFound()
-rename three methods in this class

* test: create deleteById_ok test methods for Service impls

-for: PharmacyServiceImpl
 MedicineServiceImpl
 PriceServiceImpl
-public
-> default package modifier
-inline request to my controller in tests
methods in PharmacyControllerTest

VolodymyrHoliev committed:
Yet another redundant blank lines removed

VolodymyrHoliev committed:
Fix data provider 'fields' query param
